src/mon/MDSMonitor.cc

```diff
@@ -1991,19 +2008,22 @@ void MDSMonitor::tick()
 {
   // make sure mds's are still alive
   // ...if i am an active leader
+
   if (!is_active()) return;
 
-  dout(10) << fsmap << dendl;
+  dout(10) << get_working_fsmap() << dendl;
 
-  bool do_propose = false;
+  if (!is_leader()) return;
+
+  auto &fsmap = get_pending_fsmap_writeable();
```

Comment:
To be perfectly honest, using `fsmap` for a pending map (and the writable version at that) makes me sweat a little; simply because the naming does not convey this is uncommitted state, and I can imagine (not from `tick`, I don't think) someone replying to a client with it at some point.

Reply:
I'll change it to `pending`.